liquidhaskell-boot/src/Language/Haskell/Liquid/Bare.hs

```diff
@@ -655,26 +661,36 @@ makeSpecRefl cfg src menv specs env name sig tycEnv = do
             e {eqName = s, eqRec = S.member s (exprSymbolsSet (eqBody e))}
         | (x, lt, e) <- xtes
         , let s = symbol x
-        ]
+        ] ++ otherAxioms
+  let asmReflEls = eqName <$> otherAxioms
+  let impAxioms  = concatMap (filter ((`notElem` asmReflEls) . eqName) . Ms.axeqs . snd) (M.toList specs)
   let sigVars  = F.notracepp "SIGVARS" $ (fst3 <$> xtes)            -- reflects
                                       ++ (fst  <$> gsAsmSigs sig)   -- assumes
                                       ++ (fst  <$> gsRefSigs sig)
-  return SpRefl
-    { gsLogicMap   = lmap
-    , gsAutoInst   = autoInst
-    , gsImpAxioms  = concatMap (Ms.axeqs . snd) (M.toList specs)
-    , gsMyAxioms   = F.notracepp "gsMyAxioms" myAxioms
-    , gsReflects   = F.notracepp "gsReflects" (lawMethods ++ filter (isReflectVar rflSyms) sigVars ++ wRefls)
-    , gsHAxioms    = F.notracepp "gsHAxioms" xtes
-    , gsWiredReft  = wRefls
-    , gsRewrites   = rwr
-    , gsRewritesWith = rwrWith
-    }
+  case anyNonReflFn of
+    Just (actSym , preSym) ->
+      let preSym' = show (val preSym) in
+      let errorMsg = preSym' ++ " must be reflected first using {-@ reflect " ++ preSym' ++ " @-}" in
+      let error = ErrHMeas (GM.sourcePosSrcSpan $ loc actSym) (pprint $ val actSym) (text errorMsg) :: Error
+      in Ex.throw error
+    Nothing -> return SpRefl
+      { gsLogicMap   = lmap
+      , gsAutoInst   = autoInst
+      , gsImpAxioms  = impAxioms
+      , gsMyAxioms   = myAxioms
+      , gsReflects   = lawMethods ++ filter (isReflectVar rflSyms) sigVars ++ (fst <$> gsAsmReflects sig) ++ wRefls
+      , gsHAxioms    = F.notracepp "gsHAxioms" $ xtes ++ asmReflAxioms
+      , gsWiredReft  = wRefls
+      , gsRewrites   = rwr
+      , gsRewritesWith = rwrWith
+      }
   where
     lawMethods   = F.notracepp "Law Methods" $ concatMap Ghc.classMethods (fst <$> Bare.meCLaws menv)
     mySpec       = M.lookupDefault mempty name specs
     rflSyms      = S.fromList (getReflects specs)
     lmap         = Bare.reLMap env
+    notInReflOnes (_, a) = not $ a `S.member` Ms.reflects mySpec
```

Comment:
Looks like this is going to prevent using assume-reflect with functions introduced via `measure`.